app/resources/v1/users.js

```diff
@@ -195,90 +230,288 @@ exports.post = function (req, res) {
+  }
 
-    try {
-      users = await User.find({})
-    } catch (err) {
+  if (!user) {
+    res.status(404).json({ status: 404, msg: 'User not found.' })
+    return
+  }
+
+  const idx = user.login_tokens.indexOf(req.loginToken)
+
+  const callingUser = await User.findOne({
+    where: { login_tokens: { [Op.contains]: [req.loginToken] } }
+  })
+
+  const isAdmin =
+    callingUser &&
+    callingUser.login_tokens &&
+    callingUser.login_tokens.indexOf &&
+    callingUser.login_tokens.indexOf(req.loginToken) !== -1
+
+  if (idx === -1 && !isAdmin) {
+    res.status(401).end()
+    return
+  }
+  user.login_tokens.splice(idx, 1)
+  User.update(user, { where: { id: user.id }, returning: true })
+    .then((result) => {
+      res.status(204).end()
+    })
+    .catch((err) => {
       logger.error(err)
-      throw new Error(ERRORS.CANNOT_GET_USER)
-    }
+      res.status(500).json({ status: 500, msg: 'Could not sign-out user.' })
+    })
+} // END function - exports.delete
 
-    if (!users) {
-      throw new Error(ERRORS.UNAUTHORISED_ACCESS)
-    }
+exports.put = async function (req, res) {
+  let body
+  try {
+    body = req.body
+  } catch (e) {
+    res.status(400).json({ status: 400, msg: 'Could not parse body as JSON.' })
+    return
+  }
+
+  const userId = req.params.user_id
+  let user
 
-    return users
+  try {
+    user = await User.findOne({ where: { id: userId } })
+  } catch (err) {
+    logger.error(err)
+    res.status(500).json({ status: 500, msg: 'Error finding user.' })
   }
 
-  const handleFindUsers = function (users) {
-    const usersArray = []
+  if (!user) {
+    res.status(404).json({ status: 404, msg: 'User not found.' })
+    return
+  }
 
-    const getUserJson = function (user) {
-      user.asJson({ auth: true }, function (err, userJson) {
-        if (err) {
-          logger.error(err)
-          res.status(500).json({ status: 500, msg: 'Could not render user JSON.' })
-          return
-        }
+  const callingUser = await User.findOne({
+    where: { login_tokens: { [Op.contains]: [req.loginToken] } }
+  })
 
-        userJson.profileImageUrl = user.profile_image_url
-        usersArray.push(userJson)
-      })
-    }
+  const isAdmin =
+    callingUser &&
+    callingUser.login_tokens &&
+    callingUser.login_tokens.indexOf &&
+    callingUser.login_tokens.indexOf(req.loginToken) !== -1
 
-    users.forEach((user) => { getUserJson(user) })
-    res.status(200).send(usersArray)
+  if (!isAdmin && user.login_tokens.indexOf(req.loginToken) === -1) {
+    res.status(401).end()
+    return
+  }
+  user.data = body.data || user.data || {}
+  User.update(user, { where: { id: user.id }, returning: true })
+    .then((result) => {
+      res.status(204).end()
+    })
+    .catch((err) => {
+      logger.error(err)
+      res
+        .status(500)
+        .json({ status: 500, msg: 'Could not update user information.' })
+    })
+} // END function - exports.put
+
+exports.logout = async function (req, res) {
```

Comment:
Is this duplicating functionality with `user_sessions.delete`? Why is this here? Why not use `user_sessions.delete`?